mantis-runtime/src/main/java/io/mantisrx/runtime/GroupToGroup.java

```diff
@@ -49,6 +50,11 @@
 
     GroupToGroup(GroupComputation<K1, T, K2, R> computation,
                  Config<K1, T, K2, R> config, Codec<T> inputCodec) {
+        this(computation, config, (Codec<K1>) Codecs.string(), inputCodec);
+    }
+
+    GroupToGroup(GroupComputation<K1, T, K2, R> computation,
```

Comment:
This actually results in a diamond inheritance. I'm thinking that avoiding `KeyValueInputStageConfig` and keeping the key encoder in `Stages` class is along with StageConfig is a good compromise. 
I'm unsure if I should make that optional or nullable, though. Thoughts?